x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings/use_latest_findings.ts

```diff
@@ -53,16 +53,34 @@ export const showErrorToast = (
   else toasts.addDanger(extractErrorMessage(error, SEARCH_FAILED_TEXT));
 };
 
-export const getFindingsQuery = ({ query, sort }: UseFindingsOptions) => ({
-  index: CSP_LATEST_FINDINGS_DATA_VIEW,
-  body: {
+export const getFindingsQuery = ({ query, sort }: UseFindingsOptions) => {
+  return {
+    index: CSP_LATEST_FINDINGS_DATA_VIEW,
     query,
-    sort: [{ [sort.field]: sort.direction }],
+    sort: getSortField(sort),
     size: MAX_FINDINGS_TO_LOAD,
     aggs: getFindingsCountAggQuery(),
-  },
-  ignore_unavailable: false,
-});
+    ignore_unavailable: false,
+  };
+};
+
+const getSortField = (sort: Sort<CspFinding>) => {
+  if (requiredSortingByPainlessScript.includes(sort.field)) {
+    return {
+      _script: {
+        type: 'string',
+        order: sort.direction,
+        script: {
+          source: `doc["${sort.field}"].value.toLowerCase()`,
+          lang: 'painless',
+        },
+      },
+    };
+  }
+  return { [sort.field]: sort.direction };
+};
+
+const requiredSortingByPainlessScript = ['rule.section', 'resource.name', 'resource.type'];
```

Comment:
Thanks for clarifying that @ofiriro3, 

I suggest we add some comments on the code to make our intention clearer

```suggestion
/**
 * By default, ES will sort keyword fields in case-sensitive format, the
 * following fields are required to have a custom sort for case-sensitive sorting.
 */
const fieldsSortingByPainlessScript = ['rule.section', 'resource.name', 'resource.type'];

/**
 * Generates Painless sorting if the given field is matched or returns default sorting
 */
const getSortField = ({ field, direction }: Sort<CspFinding>) => {
  if (fieldsSortingByPainlessScript.includes(field)) {
    return {
      _script: {
        type: 'string',
        order: direction,
        script: {
          source: `doc["${field}"].value.toLowerCase()`,
          lang: 'painless',
        },
      },
    };
  }
  return { [field]: direction };
};
```